packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,891 @@
+ * @param cb {Function} - Callback to trigger when the "transition" has ended.
+ */
+export const guaranteedTransitionend = (
+    el: HTMLElement,
+    action: () => void,
+    cb: () => void
+): void => {
+    const cleanup = (): void => {
+        el.removeEventListener('transitionrun', handleTransitionrun);
+        el.removeEventListener('transitionend', handleTransitionend);
+        cb();
+    };
+    let guarantee2: number;
+    let guarantee3: number;
+    // WebKit fires `transitionrun` a little earlier, so the inner/outer relationship
+    // here allows WebKit to be caught, but doesn't remove the animation listener until
+    // after it would have fired in Chromium.
+    const guarantee1 = requestAnimationFrame(() => {
+        guarantee2 = requestAnimationFrame(() => {
+            guarantee3 = requestAnimationFrame(() => {
+                cleanup();
+            });
+        });
+    });
+    const handleTransitionend = (event: TransitionEvent): void => {
+        if (event.propertyName === 'visibility') {
+            // Ignore "visibility" transitions because they often happen before/after a
+            // larger transition and don't represent the overall transition duration.
+            return;
+        }
+        cleanup();
+    };
+    const handleTransitionrun = (event: TransitionEvent): void => {
+        if (event.propertyName === 'visibility') {
+            // Ignore "visibility" transitions because they often happen before/after a
+            // larger transition and don't represent the overall transition duration.
+            return;
+        }
+        cancelAnimationFrame(guarantee1);
+        cancelAnimationFrame(guarantee2);
+        cancelAnimationFrame(guarantee3);
+        el.removeEventListener('transitionrun', handleTransitionrun);
+        el.addEventListener('transitionend', handleTransitionend);
+    };
+    el.addEventListener('transitionrun', handleTransitionrun);
+    action();
+};
+
+export class OverlayBase extends SpectrumElement {
+    static override styles = [styles];
+
+    @property({ type: Boolean })
+    delayed = false;
+
+    @query('.dialog')
+    dialogEl!: HTMLDialogElement & {
+        showPopover(): void;
+        hidePopover(): void;
+    };
+
+    @property({ type: Boolean })
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    set disabled(disabled: boolean) {
+        this._disabled = disabled;
+        if (disabled) {
+            if (this.hasNonVirtualTrigger) {
+                this.unbindEvents(this.triggerElement as HTMLElement);
+            }
+            this.wasOpen = this.open;
+            this.open = false;
+        } else {
+            this.bindEvents();
+            this.open = this.open || this.wasOpen;
+            this.wasOpen = false;
+        }
+    }
+
+    private _disabled = false;
+
+    protected dispose = noop;
+
+    @queryAssignedElements({
+        selector: ':not([slot="longpress-describedby-descriptor"])',
```

Comment:
For posterity, maybe a single line comment explaining this? Pure guess here:

```suggestion
        // the one type of child that we expect an overlay to have that can't be opened & presented is slotted
        // into the longpress-describedby-descriptior slot for accessibility
        selector: ':not([slot="longpress-describedby-descriptor"])',
```

Reply:
Seems we might not actually need that. 🤞🏼